### DISCUSSION TITLE
Deployment issue

### DISCUSSION THREAD

**Discussion by SmegDan** [eritislami/evobot#1341] [General]
Hello all,

I'm getting the below error when trying to deploy on Sparkedhost.

TSError: ⨯ Unable to compile TypeScript:
utils/i18n.ts:1:18 - error TS7016: Could not find a declaration file for module 'i18n'. '/home/container/node_modules/i18n/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/i18n` if it exists or add a new declaration (.d.ts) file containing `declare module 'i18n';`
1 import i18n from "i18n";
                   ~~~~~~
utils/i18n.ts:15:24 - error TS7006: Parameter 'msg' implicitly has an 'any' type.
15   logWarnFn: function (msg) {
                          ~~~
utils/i18n.ts:19:25 - error TS7006: Parameter 'msg' implicitly has an 'any' type.
19   logErrorFn: function (msg) {
                           ~~~
utils/i18n.ts:23:27 - error TS7006: Parameter 'locale' implicitly has an 'any' type.
23   missingKeyFn: function (locale, value) {
                             ~~~~~~
utils/i18n.ts:23:35 - error TS7006: Parameter 'value' implicitly has an 'any' type.
23   missingKeyFn: function (locale, value) {
                                     ~~~~~

Any information re. a fix for this?

**Comment by SmegDan** [comment]
My servers startup command is set to the below. Could this be the reason?

if [[ -d .git ]]; then git pull; fi; if [[ ! -z ${NODE_PACKAGES} ]]; then npm install ${NODE_PACKAGES}; fi; if [ -f /home/container/package.json ]; then  npm install --production; fi; npx ts-node /home/container/index.ts

**Reply by eritislami** [reply]
Not sure about your specific set up, but after installing packages, you can run `npm run build` this will create a `/dist` folder with transpiled javascript so you don't have to run typescript after building, this should make it easier for you

After that you can run the bot with `node ./dist/index.js`

**Reply by SmegDan** [reply]
Perfect, I shall give it a try. Unfortunately, the host I am with doesn't allow for manual changing of the startup command so I have submitted a ticket to have this changed to your suggestion of `npm run build`. Hopefully this will solve my issue and then once I see if this has solved it I will ask them to then change it to `node ./dist/index..js`

Thank you again

**Reply by eritislami** [reply]
If it's just one command then you can use `npm run prod` to build & run at the same time

```js
"scripts": {
    "dev": "nodemon index.ts",
    "start": "ts-node index.ts",
    "build": "tsc --build",
    "prod": "npm run build && node ./dist/index.js",
    "commit": "cz"
  },
```

**Reply by SmegDan** [reply]
Just need to thank you @eritislami . Your suggestion worked perfectly!
